wrapanapi/systems/msazure.py

```diff
@@ -1077,6 +1077,35 @@ def list_compute_images(self):
         return self.resource_client.resources.list(
             filter="resourceType eq 'Microsoft.Compute/images'")
 
+    def list_compute_images_by_resource_group(self, resource_group=None, free_images=None):
+        """
+        Args:
+            resource_group (str): Name of the resource group
+            free_images (bool): Whether to collect image which do not have any resource(VM) linked to it
+        """
+        resource_group = resource_group or self.resource_group
+        image_list = list(self.resource_client.resources.list(filter=f"resourceType eq 'Microsoft.Compute/images' and resourceGroup eq '{resource_group}'"))
+
+        if not free_images:
+            return image_list
+
+        vm_list = self.list_vms(resource_group=resource_group)
+        if len(vm_list) == 0:
+            return image_list
+
+        image_used_by_vm = []
+        for vm_name in vm_list:
+            image_used_by_vm.append(self.compute_client.virtual_machines.get(resource_group_name=resource_group, vm_name=vm_name).storage_profile.image_reference.id)
+
+        images_with_no_resources = []
+        for image in image_list:
+            if image.id not in image_used_by_vm:
+                images_with_no_resources.append(image)
+
+        return images_with_no_resources
```

Comment:
Set operations are more efficient and scalable solution in my view.
```suggestion
        image_used_by_vm = set()
        for vm_name in vm_list:
            image_used_by_vm.add(self.compute_client.virtual_machines.get(resource_group_name=resource_group, vm_name=vm_name).storage_profile.image_reference.id)

        images_with_no_resources = set(image_list) - image_used_by_vm

    return list(images_with_no_resources)
```

Reply:
done

Reply:
image_list and image_used_by_vm don't have the same content. image_list has the list of images as an object whereas image_used_by_vm is the IDs of the images.

Reply:
Ah, you are right. This is not worth re-working to use set operations